### ISSUE TITLE
`matrix_rep(::perm{BigInt})` broken with julia master

### ISSUE THREAD

**Issue by thofma** [Nemocas/AbstractAlgebra.jl#348]
```julia
julia> using AbstractAlgebra

julia> G = PermutationGroup(BigInt(10))
Permutation group over 10 elements

julia> a = G([2, 3, 5, 4, 6, 7, 1, 9, 10, 8])
(1,2,3,5,6,7)(8,9,10)

julia> matrix_repr(a)
ERROR: MethodError: no method matching typemax(::Type{BigInt})
Closest candidates are:
  typemax(::Type{Bool}) at bool.jl:7
  typemax(::Type{Int8}) at int.jl:666
  typemax(::Type{UInt8}) at int.jl:668
  ...
Stacktrace:
 [1] sparse!(::Array{BigInt,1}, ::Array{BigInt,1}, ::Array{BigInt,1}, ::Int64, ::Int64, ::typeof(+), ::Array{BigInt,1}, ::Array{BigInt,1}, ::Array{BigInt,1}, ::Array{BigInt,1}, ::Array{BigInt,1}, ::Array{BigInt,1}, ::Array{BigInt,1}) at /home/thofmann/julia/julia/usr/share/julia/stdlib/v1.3/SparseArrays/src/sparsematrix.jl:706
 [2] sparse(::Array{BigInt,1}, ::Array{BigInt,1}, ::Array{BigInt,1}, ::Int64, ::Int64, ::Function) at /home/thofmann/julia/julia/usr/share/julia/stdlib/v1.3/SparseArrays/src/sparsematrix.jl:564
 [3] sparse at /home/thofmann/julia/julia/usr/share/julia/stdlib/v1.3/SparseArrays/src/sparsematrix.jl:753 [inlined]
 [4] sparse(::Array{BigInt,1}, ::Array{BigInt,1}, ::Array{BigInt,1}) at /home/thofmann/julia/julia/usr/share/julia/stdlib/v1.3/SparseArrays/src/sparsematrix.jl:749
 [5] matrix_repr(::perm{BigInt}) at /home/thofmann/.julia/dev/AbstractAlgebra/src/generic/PermGroups.jl:715
 [6] top-level scope at REPL[5]:1

julia> versioninfo()
Julia Version 1.3.0-DEV.466
Commit 8d4f6d24c0 (2019-06-28 18:40 UTC)
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
```
Can you have a look @kalmarek? They must have changed something in the sparse matrix package.

**Comment by kalmarek** [comment]
@thofma this seems like a bug (or design choice) in the new SparseMatrixCSC constructors;
A direct call to constructor will work:
```julia
matrix_repr(a::perm{T}) where T = SparseMatrixCSC(length(a.d), length(a.d), collect(T, 1:length(a.d)+1), a.d, ones(T,length(a.d)))
```

btw. can we also add 1.1 and 1.2 besides nightly to travis?

**Comment by thofma** [comment]
Yes, good idea. I will make a PR (also incorporating the sparse matrix adjustment).

**Comment by thofma** [comment]
Tests are failing with
```julia
matrix_repr(a::perm{T}) where T = SparseMatrixCSC(length(a.d), length(a.d), collect(T, 1:length(a.d)+1), a.d, ones(T,length(a.d)))
```

**Comment by kalmarek** [comment]
I see, I probably switched from action on columns to action on rows (or the other way round)

do You have a branch I can commit to?

**Comment by thofma** [comment]
Sorry, no branch. Just add it to #346.

**Comment by thofma** [comment]
We are currently trying to get AbstractAlgebra working on julia master. Do you have an ETA for a fix? I would be in favor to mark it turn it into a `@test_broken` for the time being.

**Comment by kalmarek** [comment]
the problem is reported:
https://github.com/JuliaLang/julia/issues/32539
and acknowledged as a problem on master:
https://github.com/JuliaLang/julia/issues/32548

`@test_broken` will fail, since test works eg. for 1.0
I can ` VERSION < v"1.3" && @test ...` if You wish

**Comment by kalmarek** [comment]
the problem is due to new logic and sanity-checks in the constructor of SparseMatrixCSC:
https://github.com/JuliaLang/julia/blob/master/NEWS.md#sparsearrays.

is `if VERSION` a reasonable solution for you?

**Comment by thofma** [comment]
Yes, see https://github.com/Nemocas/AbstractAlgebra.jl/pull/360 :)

**Comment by wbhart** [comment]
It's fine, so long as it is not a bug in Julia that we need to report and a
problem we can work around so that all our code works in master.

**Comment by kalmarek** [comment]
@thofma, ok, please merge #360 first and then I can reschedule CI for #346

**Comment by goggle** [comment]
I have just opened a PR (https://github.com/JuliaLang/julia/pull/32631) that restores the old behavior of `sparse` in the Julia stdlib. That should solve this issue as well.

**Comment by wbhart** [comment]
Thanks for letting us know @goggle

**Comment by JeffBezanson** [comment]
This should be fixed now. Could you check to make sure?

**Comment by kalmarek** [comment]
It is fixed on f2513a8ca6 (1.3.0-alpha.1), thanks!

**Comment by rfourquet** [comment]
Shouldn't this issue be closed?
